controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -360,7 +352,7 @@ func (r *DataScienceClusterReconciler) apply(ctx context.Context, dsc *dscv1.Dat
 	}
 
 	managementStateAnn, exists := obj.GetAnnotations()[annotations.ManagementStateAnnotation]
-	if exists && managementStateAnn == string(operatorv1.Removed) {
+	if exists && (managementStateAnn == string(operatorv1.Removed) || managementStateAnn == "Unknown") {
```

Comment:
When is this case happening ? I wonder if we should only handle `Managed` and `Removed`, and return an error or skip otherwise ?

Reply:
we can set it in DSC as
```
  components:
    codeflare: {}
```
which is the same
```
  components:
    codeflare:
        managementState: Removed
```

Reply:
currently we tell user, do not set managementState in the component is the same as set it to Removed.
to make it easier i always create DSC with 
```
spec:
  components:
    modelmeshserving: 
         managementState: Managed
```
only ;D , then i found other new components get created

Reply:
so I guess the mapping could be done in the components' `NewCRObject` ?

Reply:
yeah, either do in all components NewCRObject(), or a central place.

thinking it out, if one day, we add Force or Unmanaged as supported case, we will need it in NewCRObject(), then DSC need logic to handle that as well.

Reply:
Now since components have `GetManagementState`, maybe we can move it to a central place.